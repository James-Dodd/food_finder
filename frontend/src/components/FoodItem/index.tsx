import React from "react"
import IFood from "../../models/food"
import styled from "styled-components"

const StyledFoodItem = styled.div`
  font-family: Calibri;
  color: red;
  font-size: 20px;
  font-weight: 900;
  border-style: solid;
  margin: 20px;
  padding: 10px;
`

interface IFoodItem {
  item: IFood
}

const FoodItem: React.FC<IFoodItem> = (props) => {
  function handleClick() {
    console.log("Button clicked")
    const requestOptions = {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        food: "hotdog",
        customerId: "53e827b5-7b35-444e-b1ac-0fedebb1adaa",
      }),
    }
    fetch("/order", requestOptions)
  }

  return (
    <StyledFoodItem>
      <p>{props.item.name}</p>
      <p>{props.item.price}</p>
      <p>{props.item.calories}</p>
      <p>{props.item.size}</p>
      <button type="button" onClick={handleClick}>
        Order
      </button>
    </StyledFoodItem>
  )
}

export default FoodItem
