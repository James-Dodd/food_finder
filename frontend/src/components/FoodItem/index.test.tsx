import React from "react"
import { shallow, mount } from "enzyme"
import FoodItem from "./index"

let item = {
  name: "burger",
  calories: 12,
  price: 51,
  size: "medium",
}

describe("Header", () => {
  it("renders shallow", () => {
    const component = shallow(<FoodItem item={item} />)
    expect(component).toMatchSnapshot()
  }),
    it("renders mount", () => {
      const component = mount(<FoodItem item={item} />)
      expect(component).toMatchSnapshot()
    })
})
