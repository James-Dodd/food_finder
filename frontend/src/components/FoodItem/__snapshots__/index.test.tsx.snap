// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders mount 1`] = `
<FoodItem
  item={
    Object {
      "calories": 12,
      "name": "burger",
      "price": 51,
      "size": "medium",
    }
  }
>
  <styled.div>
    <div
      className="sc-AxjAm klYsxZ"
    >
      <p>
        burger
      </p>
      <p>
        51
      </p>
      <p>
        12
      </p>
      <p>
        medium
      </p>
      <button
        onClick={[Function]}
        type="button"
      >
        Order
      </button>
    </div>
  </styled.div>
</FoodItem>
`;

exports[`Header renders shallow 1`] = `
<styled.div>
  <p>
    burger
  </p>
  <p>
    51
  </p>
  <p>
    12
  </p>
  <p>
    medium
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    Order
  </button>
</styled.div>
`;
